### PR TITLE
LPD-31949 Apply digest on PBKDF2

### PR DESCRIPTION
**Pull request by marianoalvarosaiz**

### CODE REVIEW

**Comment by alvarosaugarlr** [comment]
ci:test:stable

**Comment by alvarosaugarlr** [comment]
ci:test:relevant

**Comment by alvarosaugarlr** [comment]
ci:test:security

**Comment by stian-sigvartsen** [comment]
Hi @alvarosaugarlr . This looks like the necessary change, however I am a bit concerned about the string comparison approach. Maybe @marianoalvarosaiz can elaborate why he chose the approach.

**Comment by marianoalvarosaiz** [comment]
@stian-sigvartsen I also tried to find a way to delegate in the library the Digest creation to abstract it somehow, but I didn't find a useful method, so I had to do it "manually" in the end.

**Comment by rafaprax** [comment]
Hey @marianoalvarosaiz, Did you try this method here https://github.com/bcgit/bc-java/blob/main/prov/src/main/java/org/bouncycastle/jcajce/provider/util/DigestFactory.java#L136 ?

As IDK the possible values of the algorithm variable, I'm not sure if it is suitable for this case. Can you check that?

Thank you

**Comment by marianoalvarosaiz** [comment]
@rafaprax, String manipulation still has to be done, but I'll take a look.

**Comment by marianoalvarosaiz** [comment]
ci:test:sf

**Comment by marianoalvarosaiz** [comment]
ci:test:relevant

**Comment by marianoalvarosaiz** [comment]
ci:test:security

File: modules/apps/portal-security/portal-security-password-encryptor-impl/src/main/java/com/liferay/portal/security/password/encryptor/internal/PBKDF2PasswordEncryptor.java [hunk 1]
```diff
@@ -134,6 +139,23 @@ public void configure(String algorithm, String encryptedPassword)
...
 			}
+
+			int indexOf = StringUtil.indexOfIgnoreCase(algorithm, "SHA");
+
+			if (indexOf < 0) {
+				return;
+			}
+
+			_digest = DigestFactory.getDigest(
+				algorithm.substring(
```
Comment (rafaprax) [inline modules/apps/portal-security/portal-security-password-encryptor-impl/src/main/java/com/liferay/portal/security/password/encryptor/internal/PBKDF2PasswordEncryptor.java:150]: @marianoalvarosaiz please remove the chaining here. After that, we can forward this. Thanks for applying the new approach.

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by marianoalvarosaiz** [comment]
@rafaprax force pushed without the method chaining.

**Comment by marianoalvarosaiz** [comment]
ci:open

**Comment by marianoalvarosaiz** [comment]
Rebased, with no additional changes, see https://github.com/liferay-appsec/liferay-portal/pull/1895

**Pull request closed without merging**